wcfsetup/install/files/lib/system/package/PackageArchive.class.php

```diff
@@ -426,13 +429,21 @@ protected function filterUpdateInstructions() {
 	 * @return	string		path to the dowloaded file
 	 */
 	public function downloadArchive() {
-		$prefix = 'package';
+		$tmpFile = FileUtil::getTemporaryFilename('package_');
 		
-		// file transfer via hypertext transfer protocol.
-		$this->archive = FileUtil::downloadFileFromHttp($this->archive, $prefix);
+		$client = HttpFactory::getDefaultClient();
+		$request = new Request('GET', $this->archive, [
+			'stream' => true,
+		]);
```

Comment:
```suggestion
		$request = new Request('GET', $this->archive);
```

Reason: Guzzle automatically reads the response into a `fopen('php://temp')` stream. This one automatically spills over to disk anyway and thus will not overflow the memory.